### PR TITLE
core: bump vllm to 0.13.0

### PR DESCRIPTION
**Pull request by rebel-eunji**
### 🚀 Summary of Changes

- This PR temporarily turns off the pre-commit hook.
- Check whether LoRA works well.

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`bug-fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -39,9 +39,9 @@ dependencies = [
     "pyyaml",
     "datasets",
     "qwen_vl_utils",
-    "vllm==0.10.2",
-    "optimum-rbln==0.10.0.post2",
-    "torch==2.8.0",
+    "vllm==0.13.0",
+    "optimum-rbln>=0.10.1a2",
```
Comment (rebel-eunji) [inline pyproject.toml]:
```suggestion
    "optimum-rbln>=0.10.1a1",
```

File: vllm_rbln/v1/core/optimum_scheduler.py [hunk 2]
```diff
@@ -135,17 +157,43 @@ def __init__(
             attn_block_size = None
         self.kv_cache_manager = RBLNKVCacheManager(
             kv_cache_config=kv_cache_config,
-            max_model_len=self.max_model_len,
+            max_model_len=self.max_num_scheduled_tokens,
```
Comment (rebel-jaehwang) [inline vllm_rbln/v1/core/optimum_scheduler.py]: looks weird

File: vllm_rbln/v1/core/optimum_block_pool.py [hunk 3]
```diff
@@ -12,48 +12,74 @@
...
+        if blocks is not None:
+            if isinstance(blocks, KVCacheBlock):
+                # Only one block is allocated.
+                # This block is allocated by the current request itself.
+                return None
+            if isinstance(blocks, dict):
+                # Multiple blocks are allocated.
+                # Return the second-to-last block.
+                block_values = blocks.values()
+                print(f"block_values: {block_values}")
```
Comment (rebel-jaehwang) [inline vllm_rbln/v1/core/optimum_block_pool.py]: print

File: vllm_rbln/model_executor/models/optimum/encoder.py [hunk 4]
```diff
@@ -42,21 +41,14 @@ def __init__(self) -> None:
     def get_supported_tasks(self) -> set[PoolingTask]:
         return {"classify", "score"}
 
-    @staticmethod
-    def _build_output(
-        all_data: Union[torch.Tensor, list[torch.Tensor]], ) -> PoolerOutput:
-        """Wrap tensor data into vLLM's PoolerOutput format."""
-        all_outputs = [PoolingSequenceGroupOutput(data) for data in all_data]
```
Comment (rebel-jonghewk) [inline vllm_rbln/model_executor/models/optimum/encoder.py]: Is `PoolingSequenceGroupOutput` deleted from vllm?

File: vllm_rbln/model_executor/models/optimum/idefics3.py [hunk 5]
```diff
@@ -129,37 +126,36 @@ def _parse_and_validate_image_input(
             return None
 
         if image_embeds is not None:
-            if not isinstance(image_embeds, (torch.Tensor, list)):
+            if not isinstance(image_embeds, torch.Tensor | list):
```
Comment (rebel-jonghewk) [inline vllm_rbln/model_executor/models/optimum/idefics3.py]: Are we deprecating py3.9?

File: vllm_rbln/model_executor/models/optimum/qwen_vl.py [hunk 6]
```diff
@@ -346,12 +302,13 @@ def _create_video_pixel_inputs(self,
                 "second_per_grid_ts is required for Qwen2.5-VL video inputs.")
         # NOTE vLLM also squeezes the second_per_grid_ts tensor
         # https://github.com/vllm-project/vllm/blob/v0.10.2/vllm/model_executor/models/qwen2_5_vl.py#L1021
-        if envs.VLLM_USE_V1:
-            # [num_videos, 1] -> [num_videos]
-            second_per_grid_ts = second_per_grid_ts.squeeze(-1)
-        else:
-            # [1, num_videos] -> [num_videos]
-            second_per_grid_ts = second_per_grid_ts.squeeze(-2)
+        # if envs.VLLM_USE_V1:
```
Comment (rebel-jonghewk) [inline vllm_rbln/model_executor/models/optimum/qwen_vl.py]: remove?

Comment (rebel-eunji) [reply vllm_rbln/model_executor/models/optimum/qwen_vl.py]: Yes I'll remove it.

File: vllm_rbln/v1/worker/rbln_worker.py [hunk 7]
```diff
@@ -431,4 +431,4 @@ def init_worker_distributed_environment(
         parallel_config.pipeline_parallel_size,
     )
 
-    ensure_kv_transfer_initialized(vllm_config)
+    ensure_kv_transfer_initialized(vllm_config)
```
Comment (rebel-eunji) [inline vllm_rbln/v1/worker/rbln_worker.py]:
```suggestion
    ensure_kv_transfer_initialized(vllm_config)

```

File: tests/v1/core/utils.py [hunk 8]
```diff
@@ -45,11 +44,9 @@ def create_scheduler(
     block_size: int = 16,
     max_model_len: Optional[int] = None,
     async_scheduling: bool = False,
-    is_torch_compile: bool = False,
```
Comment (huijjj) [inline tests/v1/core/utils.py]: Great. We'll manage the tests for the torch.compile path separately, without relying on the optimum path codes. So you don’t need to worry about the torch.compile side in tests!
